File: app_mobile/lib/core/models/product.dart
class Product {
  final String id;
  final String barcode;
  final String name;
  final String? brand;
  final String? categoryId;
  final String? imageUrl;
  final int? daysAfterOpening;
  final String source;

  Product({
    required this.id,
    required this.barcode,
    required this.name,
    this.brand,
    this.categoryId,
    this.imageUrl,
    this.daysAfterOpening,
    required this.source,
  });

  factory Product.fromJson(Map<String, dynamic> json) => Product(
        id: json['id'] as String,
        barcode: json['barcode'] as String,
        name: json['name'] as String,
        brand: json['brand'] as String?,
        categoryId: json['categoryId'] as String?,
        imageUrl: json['imageUrl'] as String?,
        daysAfterOpening: json['daysAfterOpening'] as int?,
        source: json['source'] as String,
      );
}

File: backend/src/main/java/com/mycosplaymanager/backend/entity/Product.java
package com.mycosplaymanager.backend.entity;

import java.time.Instant;
import java.util.UUID;

import org.hibernate.annotations.UuidGenerator;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/** Catalogo crowdsourced condiviso tra tutti i team: chi trova un barcode assente lo aggiunge. */
@Entity
@Table(name = "products")
@Getter
@Setter
@NoArgsConstructor
public class Product {

    @Id
    @UuidGenerator
    private UUID id;

    @Column(nullable = false, unique = true)
    private String barcode;

    @Column(nullable = false)
    private String name;

    private String brand;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(name = "image_url")
    private String imageUrl;

    /** Giorni entro cui va consumato dopo l'apertura (proprieta' del prodotto, non del singolo item). */
    @Column(name = "days_after_opening")
    private Integer daysAfterOpening;

    @Column(nullable = false)
    private String source = "USER";

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @PrePersist
    void onCreate() {
        this.createdAt = Instant.now();
    }
}
